docs/capi.rst

```diff
@@ -1908,3 +1908,147 @@ Memory Management:
    nrn_object_unref(obj);  // Decrement reference count
+
+
+
+Runnable example with compilation instructions
+----------------------------------------------
+
+A Hodgkin-Huxley NEURON simulation in C++17
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we stimulate the cell with a :class:`IClamp` at time 1 ms.
+
+.. code-block:: c++
+
+    #include "neuronapi.h"
+
+    #include <array>
+    #include <cstdlib>
+    #include <cstring>
+    #include <iostream>
+
+    using std::cout;
+    using std::endl;
+    using std::pair;
+
+    extern "C" void modl_reg(){/* No modl_reg */};
+
+    int main(void) {
+        static std::array<const char*, 4> argv = {"hh_sim", "-nogui", "-nopython", nullptr};
+        nrn_init(3, argv.data());
+
+        // load the stdrun library
+        char* temp_str = strdup("stdrun.hoc");
+        nrn_str_push(&temp_str);
+        nrn_function_call(nrn_symbol("load_file"), 1);
+        nrn_double_pop();
+        free(temp_str);
+
+        // topology
+        Section* soma = nrn_section_new("soma");
+        nrn_nseg_set(soma, 3);
+
+        // define soma morphology with two 3d points
+        nrn_section_push(soma);
+        for (double x: {0, 0, 0, 10}) {
+            nrn_double_push(x);
+        }
+        nrn_function_call(nrn_symbol("pt3dadd"), 4);
+        nrn_double_pop();  // pt3dadd returns a number
+        for (double x: {10, 0, 0, 10}) {
+            nrn_double_push(x);
+        }
+        nrn_function_call(nrn_symbol("pt3dadd"), 4);
+        nrn_double_pop();  // pt3dadd returns a number
+
+        // ion channels
+        nrn_mechanism_insert(soma, nrn_symbol("hh"));
+
+        // current clamp at soma(0.5)
+        nrn_double_push(0.5);
+        Object* iclamp = nrn_object_new(nrn_symbol("IClamp"), 1);
+        for (const auto& [property, value] : {pair{"amp", 0.3}, pair{"del", 1.0}, pair{"dur", 0.1}}) {
+            nrn_property_set(iclamp, property, value);
+        }
+
+        // setup recording
+        Object* v = nrn_object_new(nrn_symbol("Vector"), 0);
+        nrn_rangevar_push(nrn_symbol("v"), soma, 0.5);
+        nrn_method_call(v, nrn_method_symbol(v, "record"), 1);
+        nrn_object_unref(nrn_object_pop());  // record returns the vector
+
+        Object* t = nrn_object_new(nrn_symbol("Vector"), 0);
+        nrn_double_ptr_push(nrn_symbol_dataptr(nrn_symbol("t")));
+        nrn_method_call(t, nrn_method_symbol(t, "record"), 1);
+        nrn_object_unref(nrn_object_pop());  // record returns the vector
+
+
+        // finitialize(-65)
+        nrn_double_push(-65);
+        nrn_function_call(nrn_symbol("finitialize"), 1);
+        nrn_double_pop();
+
+        // continuerun(5)
+        nrn_double_push(5);
+        nrn_function_call(nrn_symbol("continuerun"), 1);
+        nrn_double_pop();
+
+        // Print CSV output: time, voltage
+        int n_points = nrn_vector_capacity(t);
+        double* t_data = nrn_vector_data(t);
+        double* v_data = nrn_vector_data(v);
```

Comment:
Since there is no call to `free` on either `t_data` or `v_data`, I think there is a memory leak here, no? Unless NEURON manages the memory.